File: model-client/src/main/java/org/modelix/model/client/IModelClient.java
package org.modelix.model.client;

import org.modelix.model.IKeyValueStore;
import org.modelix.model.api.IIdGenerator;
import org.modelix.model.lazy.IDeserializingKeyValueStore;

public interface IModelClient extends IKeyValueStore {
  int getClientId();
  IIdGenerator getIdGenerator();
  IDeserializingKeyValueStore getStoreCache();
  IKeyValueStore getAsyncStore();
}

File: model-client/src/main/kotlin/org/modelix/model/client/IModelClient.kt
package org.modelix.model.client

import org.modelix.model.IKeyValueStore
import org.modelix.model.api.IIdGenerator
import org.modelix.model.lazy.IDeserializingKeyValueStore

interface IModelClient : IKeyValueStore {
    val clientId: Int
    val idGenerator: IIdGenerator?
    val storeCache: IDeserializingKeyValueStore?
    val asyncStore: IKeyValueStore?
}
